src/Sylius/Component/Product/Model/Product.php

```diff
@@ -474,4 +502,22 @@ protected function createTranslation()
     {
         return new ProductTranslation();
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    private function getAttributeInDifferentLocale(ProductAttributeValueInterface $attributeValue, $localeCode)
+    {
+        if (!$this->hasAttributeByCodeAndLocale($attributeValue->getCode(), $localeCode)) {
+            return $attributeValue;
+        }
+
+        $attributeValueInDifferentLocale = $this->getAttributeByCodeAndLocale($attributeValue->getCode(), $localeCode);
+        if ('' === $attributeValueInDifferentLocale->getValue()
+            || null === $attributeValueInDifferentLocale->getValue()) {
```

Comment:
empty($attributeValueInDifferentLocale->getValue()) will be equivalent to this if. And we will remove an `or` condition